demos/a2a_llama_stack/README.md

```diff
@@ -1,127 +1,266 @@
-# Integrating a Custom Agent with Google A2A on **Llama Stack**
+# Running Your Llama Stack Agent with Google A2A
 
-This guide walks you through running a custom agent on **Llama Stack** using Google’s **Agent‑to‑Agent (A2A)** protocol.
+Welcome! This guide provides comprehensive instructions for setting up and running a custom agent on **Llama Stack**, leveraging Google’s **Agent‑to‑Agent (A2A)** communication protocol. Follow these steps to make your agent operational.
 
 ---
 
-## 🛠 Prerequisites
+## Overview
 
-- **Python 3.8 or newer**
-- **`pip`** (Python package manager)
-- A **Llama Stack** inference server running and reachable
+By completing this guide, you will accomplish the following:
+1.  Setting up your development environment.
+2.  Download the requisite code repositories.
+3.  Install all necessary dependencies.
+4.  Configure connection details for your Llama Stack inference server.
+5.  Launch the A2A agent server(s).
+6.  Execute a client application to dispatch tasks to your agent(s).
 
 ---
 
-## 1 — Download the Code
+## Prerequisites
+
+Before commencing, please ensure the following components are installed and accessible:
+* **Python 3.8 or newer**
+* **`pip`** (Python package manager)
+* A **Llama Stack** inference server, running and reachable from your machine.
+
+---
+
+## Setup Instructions
+
+Follow these steps to prepare your environment and the application code.
+
+### 1. Download the Required Code
+
+Begin by cloning two Git repositories: the Llama Stack demos and the Google A2A examples.
 
 ```bash
-# Clone the Llama Stack demos
+# Clone the Llama Stack demos repository
 git clone https://github.com/opendatahub-io/llama-stack-demos.git
 
-# Clone the Google A2A examples
+
+# Clone the Google A2A examples repository
 git clone https://github.com/google/A2A.git
```

Comment:
The samples directory has been moved to https://github.com/google-a2a/a2a-samples/tree/main/samples/python